lib/indexeddb.js

```diff
@@ -0,0 +1,118 @@
+// Implemented by window & worker
+declare interface IDBEnvironment {
+  indexedDB: IDBFactory;
+}
+
+type IDBDirection = 'next' | 'nextunique' | 'prev' | 'prevunique';
+
+// Implemented by window.indexedDB & worker.indexedDB
+declare interface IDBFactory {
+  open(name: string, version?: number): IDBOpenDBRequest;
+  deleteDatabase(name: string): IDBOpenDBRequest;
+  cmp(a: any, b: any): -1|0|1;
+}
+
+declare interface IDBRequest extends EventTarget {
+  result: IDBObjectStore;
+  error: Error;
+  source: ?(IDBIndex | IDBObjectStore | IDBCursor);
+  transaction: IDBTransaction;
+  readyState: 'pending'|'done';
+  onerror: (err: any) => void;
+  onsuccess: (e: any) => void;
+}
+
+declare interface IDBOpenDBRequest extends IDBRequest {
+  onblocked: (e: any) => void;
+  onupgradeneeded: (e: any) => void;
+}
+
+declare interface IDBDatabase extends EventTarget {
+  close(): void;
+  createObjectStore(name: string, options?: {
+    keyPath?: string|string[],
+    autoIncrement?: bool
+  }): IDBObjectStore;
+  deleteObjectStore(name: string): void;
+  transaction(storeNames: string|string[], mode?: 'readonly'|'readwrite'|'readwriteflush'): IDBTransaction;
+  name: string;
+  version: number;
+  objectStoreNames: string[];
+  onabort: (e: any) => void;
+  onerror: (e: any) => void;
+  onversionchange: (e: any) => void;
+}
+
+declare interface IDBTransaction extends EventTarget {
+  abort(): void;
+  objectStore(name: string): IDBObjectStore;
+  db: IDBDatabase;
+  mode: 'readonly'|'readwrite'|'versionchange';
+  objectStoreNames: string[];
+  error: Error;
+  onabort: (e: any) => void;
+  onerror: (e: any) => void;
+  oncomplete: (e: any) => void;
+}
+
+declare interface IDBObjectStore {
+  add(value: any, key?: any): IDBRequest;
+  clear(): IDBRequest;
+  delete(key: any): IDBRequest;
+  get(key: any): IDBRequest;
+  getAll(query?: any|IDBKeyRange, count?: number): IDBRequest;
+  getAllKeys(query?: IDBKeyRange, count?: number): IDBRequest;
+  createIndex(indexName: string, keyPath: ?(string|string[]), optionalParameter?: {
```

Comment:
I don't think keyPath is nullable ([spec](https://www.w3.org/TR/IndexedDB/#idl-def-IDBObjectStore))